src/activity_handlers.cpp

```diff
@@ -262,6 +262,9 @@ void set_up_butchery( player_activity &act, player &u, butcher_type action )
         return;
     }
 
+    item corpse_item = items[act.index];
+    const mtype *corpse = corpse_item.get_mtype();
```

Comment:
Could change to `const mtype& corpse = *(corpse_item.get_mtype());` so you don't pay for dereferencing it through the rest of the function. Additionally, since it has been moved up from its previous location use it.